packages/core/src/browser/frontend-application.ts

```diff
@@ -263,10 +273,9 @@ export class FrontendApplication {
     protected revealShell(host: HTMLElement): Promise<void> {
         const startupElem = this.getStartupIndicator(host);
         if (startupElem) {
-            return new Promise(resolve => {
+            const doRevealShell = () => new Promise<void>(resolve => {
```

Comment:
Since we're already adding several measurements to the `start` method, which is where this is called, it seems more consistent (and easier) to add the measurement there? I'm not sure how often we enter the `else` here - in principle, it shouldn't be often.

Reply:
Indeed, this was too cute by half. Thanks for the insight about the else case!